src/test/measuretime.h

```diff
@@ -3,12 +3,19 @@
 #include <chrono>
 #include <iomanip>
 #include <iostream>
+#include <sstream>
 
-#define DO_TIME(name, code) \
-  { \
-    auto start__ = std::chrono::high_resolution_clock::now(); \
-    code auto finish__ = std::chrono::high_resolution_clock::now(); \
-    auto diff__ = finish__ - start__; \
-    std::cout << name << ": " << std::setw(12) << diff__.count() << " ns" \
-              << std::endl; \
+class MeasureTime : public std::stringstream
```

Comment:
I'm not quite sure how to feel about this.  I think... it's cute, but it seems very odd to both extend `std::stringstream` and have such striking side effects in the destructor.  Still, it does make for rather EDSL-esque use.

Reply:
I was trying to minimise the change from the original macro.